pkg/virtctl/create/vm/vm.go

```diff
@@ -668,6 +694,13 @@ func (c *createVM) cloudInitConfig(cmd *cobra.Command, vm *v1.VirtualMachine) er
 		return err
 	}
 
+	// Do not add cloud-init config if it was not requested and only the header is present
+	if !c.cmd.Flags().Changed(CloudInitFlag) &&
+		((src.CloudInitNoCloud != nil && src.CloudInitNoCloud.UserData == CloudInitConfigHeader && src.CloudInitNoCloud.NetworkDataBase64 == "") ||
+			(src.CloudInitConfigDrive != nil && src.CloudInitConfigDrive.UserData == CloudInitConfigHeader && src.CloudInitConfigDrive.NetworkDataBase64 == "")) {
+		return nil
+	}
```

Comment:
nit - This should be extracted into one or more helpers IMHO. I don't see anything asserting this behaviour as well?

Reply:
Right, addressed.